tests/functional/u/unspecified_default_for_next.py

```diff
@@ -0,0 +1,35 @@
+"""Warnings for using next() without specifying a default value."""
+
+next((i for i in (1, 2)), None)
+next(i for i in (1, 2))  # [unspecified-default-for-next]
+var = next(i for i in (1, 2))  # [unspecified-default-for-next]
+
+try:
+    next(i for i in (1, 2))
+except StopIteration:
+    pass
+
+try:
+    next(i for i in (1, 2))  # [unspecified-default-for-next]
+except ValueError:
+    pass
+
+try:
+    next(i for i in (1, 2))
+except (ValueError, StopIteration):
+    pass
+
+try:
+    next(i for i in (1, 2))
+except ValueError:
+    pass
+except StopIteration:
+    pass
+
```

Comment:
We have a lot of example in astroid, but we can rely on the primer. Let's just handpick one:

```suggestion


def _unpack_keywords(self, keywords, context=None):
    for name, value in keywords:
        try:
            inferred = next(value.infer(context=context))
        except StopIteration:
            continue


```